### ISSUE TITLE
M2.2.x: Products not being removed from website via REST API

### ISSUE THREAD

**Issue by thomvanderboon** [magento/magento2#12190]
Under M2.1.x I was able to add and remove a product to a website using the REST API. Under M2.2.x adding a product to a website via RESTworks, but removing the product via REST from the website does not work.

### Preconditions
1. Ubuntu 16.04 LTS
2. PHP 7.0.22
3. Extra user in database to handle rest via tokens

### Steps to reproduce
1. Create a product and a extra website 
2. Run the following script:

```
<?php

$sku = "5000001_STUK"; // Change your SKU to match a product in your database

//Authentication rest API magento2.Please change url accordingly your url
$adminUrl='http://shop.domain.com/rest/V1/integration/admin/token';
$ch = curl_init();
$data = array("username" => "rest_user", "password" => "Some_secret_password");
$data_string = json_encode($data);
$ch = curl_init($adminUrl);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
curl_setopt($ch, CURLOPT_POSTFIELDS, $data_string);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    'Content-Type: application/json',
    'Content-Length: ' . strlen($data_string))
);
$token = curl_exec($ch);
$token=  json_decode($token);
echo "Token: " .$token . "\n";
curl_close($ch);

//Use above token into header
$headers = array("Authorization: Bearer $token");


// Remove product from website id 1

$headers = array("Authorization: Bearer $token","Content-Type: application/json");
$requestUrl="http://shop.domain.com/index.php/rest/V1/products/".$sku."/websites/1";
$ch = curl_init();
$ch = curl_init($requestUrl);
//curl_setopt($ch, CURLOPT_POSTFIELDS, $data_string);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "DELETE");
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$curl_result = curl_exec($ch);
$responseInfo = curl_getinfo($ch);
$httpResponseCode = $responseInfo['http_code'];
echo $httpResponseCode;
print_r($curl_result);
curl_close($ch);

// Add product to wesite id 2

$sampleProductData = array(
  'sku' => $sku,
  'website_id' => 2,
  );
$data_string = json_encode(array('productWebsiteLink' => $sampleProductData));

$requestUrl="http://shop.domain.com/index.php/rest/V1/products/".$sku."/websites";
$ch = curl_init();
$ch = curl_init($requestUrl);
curl_setopt($ch, CURLOPT_POSTFIELDS, $data_string);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$curl_result = curl_exec($ch);
print_r($curl_result);
curl_close($ch);
?>
```

### Expected result
1. Product is removed from website id 1 and added to website id 2

### Actual result
1. Product NOT removed from website id 1, Product is correctly added to website id 2

Tested against my backup 2.1.9. Above script works perfect on M2.1.9

**Comment by afirlejczyk** [comment]
I'd like to work on this one.

**Comment by magento-engcom-team** [comment]
@thomvanderboon, thank you for your report.
We've created internal ticket(s) MAGETWO-84161 to track progress on the issue.

**Comment by afirlejczyk** [comment]
I want to give it a try so please assign it to me.

**Comment by p-bystritsky** [comment]
Issue already fixed in 2.2-develop and 2.3-develop.
